indico/modules/events/registration/api.py

```diff
@@ -7,19 +7,106 @@
 
 from flask import jsonify, request, session
 from sqlalchemy.orm import joinedload
+from webargs import fields
 from werkzeug.exceptions import BadRequest, Forbidden
 
 from indico.core import signals
+from indico.modules.events import Event
 from indico.modules.events.controllers.base import RHProtectedEventBase
-from indico.modules.events.models.events import Event
-from indico.modules.events.registration.models.registrations import RegistrationState
+from indico.modules.events.registration.models.forms import RegistrationForm
+from indico.modules.events.registration.models.registrations import Registration, RegistrationState
 from indico.modules.events.registration.util import build_registration_api_data, build_registrations_api_data
-from indico.web.rh import RH, oauth_scope
+from indico.web.args import use_kwargs
+from indico.web.rh import RH, cors, json_errors, oauth_scope
+
+
+@json_errors
+@cors
+@oauth_scope('registrants')
+class RHAPICheckinBase(RHProtectedEventBase):
+    """Base class for the Check-in API."""
+
+    def _check_access(self):
+        RHProtectedEventBase._check_access(self)
+
+    def _process_args(self):
+        RHProtectedEventBase._process_args(self)
+
+
+class RHAPIEvent(RHAPICheckinBase):
+    """Manage a single event."""
+
+    def _process_GET(self):
```

Comment:
usually we just use `_process` when we don't need to run different logic for different methods